inference-engine/src/vpu/graph_transformer/src/stages/dynamic_shape_resolver.cpp

```diff
@@ -50,6 +50,11 @@ void FrontEnd::parseDSR(const Model& model, const ie::CNNLayerPtr& layer, const
         // Create the second output with shape in case of dynamic output
         const auto& shapeOutput = model->addOutputData(dataOutput->name() + "@shape", shape->desc());
 
+        bindData(shapeOutput, shape->origData());
+        for (const auto& shapeConsumerEdge : shape->consumerEdges()) {
+            model->replaceStageInput(shapeConsumerEdge, shapeOutput);
+        }
+
```

Comment:
I see the case, that data object with several consumers, first we parse one of them, so we don't need to replace anything, becuase other are not parsed yet (that meet naive assumptions about frontend parsing - there is no consumers).

But then, we parse other consumers and there replacing them matters. Do you think we missed to do the same (replacing consumers) in other places?

Reply:
We may miss something, but `parseDSR` is the first parse method which use `replaceStageOutput`.
By the way, I guess, we have to do the same after `model->replaceStageOutput(dataProducerEdge, dataOutput);` above

Reply:
Probably, @AndrewBakalinIntel please take a look at that place

Reply:
> By the way, I guess, we have to do the same after `model->replaceStageOutput(dataProducerEdge, dataOutput);` above

Although this will be an unnecessary precaution. Input data must have no consumers except DSR

Reply:
There can be a case when `data` that comes to DSR have some other consumers besides DSR, but I would treat it rather a bug than a case that we need to support, because the data that comes to DSR is dynamic, and second input is its shape, so everyone who wants to consume this dynamic tensor should be a consumer of DSR's output, otherwise it seems like someone wants to process tensor with garbage elements, which IMO is incorrect.

Reply:
@AndrewBakalinIntel, I agree with you regarding the input `data`. We can add assertion for this case. 
Anyway, `shape` can be input for current DSR and input for other stage (Broadcast for example). And that stage may be parsed earlier.

Reply:
Yes, this is what was missed in the current implementation.

Reply:
Should it be resolved?

Reply:
I suppose, yes